From: "J. R. Okajima" <hooanon05@yahoo.co.jp>
Date: Wed, 17 Jul 2013 01:04:06 +0900
Subject: [48/83] aufs: bugfix, clear S_AUTOMOUNT
Origin: http://sourceforge.net/p/aufs/aufs3-standalone/ci/48e9fd849cb01625b73cc56a3837f095a720a44b

Aufs inode should not copy-up the inode S_AUTOMOUNT flag from its branch
inode. The S_AUTOMOUNT flag was introduced in linux-2.6.38, so this is a
long-lived bug.
Currently (linux-3.11-rc1 mainline), this flag is set only for
- Andrew File System support (AFS)
- CIFS (CIFS_FATTR_DFS_REFERRAL case only)
- NFS (NFS_ATTR_FATTR_MOUNTPOINT or NFS_ATTR_FATTR_V4_REFERRAL case only)
and I have never tested such branches.
Found and reported by Ken Elkabany and Aaron Staley who are using
OpenAFS (out of mainline) as an aufs branch.

Reported-by: Ken Elkabany <ken@picloud.com>
Reported-by: Aaron Staley <aaron@picloud.com>
Signed-off-by: J. R. Okajima <hooanon05@yahoo.co.jp>
---
 fs/aufs/cpup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/aufs/cpup.c
+++ b/fs/aufs/cpup.c
@@ -27,7 +27,7 @@
 void au_cpup_attr_flags(struct inode *dst, struct inode *src)
 {
 	const unsigned int mask = S_DEAD | S_SWAPFILE | S_PRIVATE
-		| S_NOATIME | S_NOCMTIME;
+		| S_NOATIME | S_NOCMTIME | S_AUTOMOUNT;
 
 	dst->i_flags |= src->i_flags & ~mask;
 	if (au_test_fs_notime(dst->i_sb))
